src/ml-service/packages/sensor_location/locate/app.py

```diff
@@ -0,0 +1,152 @@
+            if polygon == {}:
+                return {'message': 'Please draw a polygon'}, 200
+            geometry = polygon["geometry"]["coordinates"]
+
+            must_have_coordinates = json_data["must_have_coordinates"]
+            if must_have_coordinates == "":
+                must_have_coordinates = None
+                return locate_helper.recommend_locations(sensor_number, must_have_coordinates, geometry)
+            else:
+                try:
+                    must_have_coordinates = ast.literal_eval(
+                        must_have_coordinates)
+                except:
+                    return {'message': 'Coordinates must be in the form [[long, lat], [long, lat]]'}, 200
+                if all(isinstance(x, list) for x in must_have_coordinates):
+                    return locate_helper.recommend_locations(sensor_number, must_have_coordinates, geometry)
+                else:
+                    return {'message': 'Coordinates must be in the form [[longitude, latitude]]'}, 200
+
+
+@app.route('/api/v1/map/savelocatemap', methods=['GET', 'POST'])
+def save_locate_map():
+    '''
+    Saves planning space
+    '''
+    if request.content_type != 'application/json':
+        error = json.dumps({'error': 'Invalid Content Type'})
+        return jsonify(error, 400)
+
+    data = request.json
+    if not all([data.get('user_id'), data.get('space_name'), data.get('plan')]):
+        error = json.dumps({'error': 'Missing field/s (title, author_id)'})
+        return jsonify(error, 400)
+
+    user_id = data['user_id']
+    space_name = data['space_name']
+    plan = data['plan']
+
+    locate_model.save_locate_map(user_id, space_name, plan)
+
+    return jsonify({"message": "Locate Plannig Space Saved Successfully", "status": 200})
+
+# get previously saved planning space by the current user
+@app.route('/api/v1/map/getlocatemap/<user_id>')
+def get_locate_map(user_id):
+    '''
+    Get saved planning space for the user
+    '''
+    documents = locate_model.get_locate_map(user_id)
+    response = []
+    for document in documents:
+        document['_id'] = str(document['_id'])
+        response.append(document)
+    data = jsonify(response)
+    return data
+
+# Update previously saved planning space
+@app.route('/api/v1/map/updatelocatemap/<space_name>', methods=['GET', 'POST'])
+def update_locate_map(space_name):
+    '''
+    updates a previously saved planning space
+
+    @param: space_name
+    @return: null
+
+    '''
+    pass
+
+# Delete previously saved planning space
+@app.route('/api/v1/map/deletelocatemap/<space_name>', methods=['GET', 'POST', 'DELETE', 'PUT'])
+def delete_locate_map(space_name):
+    '''
+    deletes a previously saved planning space
+    @param: space_name
+    @return: null
+    '''
+    if request.method == 'DELETE':
+        if space_name is not None:
+            db_response = locate_model.delete_locate_map(space_name)
+            if db_response.deleted_count == 1:
```

Comment:
Could we be able to have some consistency in the structure of our response messages? Noticed that for the POST of creating a new planning space, we have {"message":"", "status":""} and the response for deleting a space comes {"message":"", "ok":""}.   Perhaps we could have: {"message":"", success: Boolean} for all?

I think I will also have to do for other services we implemented - "note to self"

Reply:
True... I'll work on it immediately. Thank you!